nil/getting-started/essentials/creating-a-wallet.mdx

```diff
@@ -0,0 +1,56 @@
+# Creating a wallet
+
+This tutorial defines how wallets work in =nil; and outlines the different ways for creating a new wallet.
+
+## Definition
+
+In =nil; a wallet is any smart contract that is able to pay for interactions with real-life users.
+
+The traditional definition of a wallet in Ethereum mostly refers to externally-owned accounts (EOAs) that are secured via private keys and natively hold information about user's balances.
+
+Wallets in =nil; differ from this definition. A =nil; wallet is always a smart contract, and it must be deployed as such. The wallet handles all the usual functionalities (e.g., authenticating requests) but it can also include complex logic such as multi-sig operations.
+
+:::info[Counterfactual wallets]
+
+It is possible to send funds to an address where a contract is supposed to be deployed. These funds will be stored at that address until the contract is finally deployed, at which point the contract will be able to access them.
+
+Funds sent to an address without a deployed contract are safe, and another party cannot claim them by deploying another contract before the owners of the funds is able to deploy their own contract.
+
+:::
+
+## New wallet creation
+
+### Via the =nil; CLI
+
+The =nil; CLI offers a quick and convenient way to create a new wallet:
```

Comment:
It is important to say that `nil_cli new-wallet` only works in devnet, as it tops up the wallet with free tokens. In real production network, the process will be different and likely involve going through an exchange to get some tokens to your address and only then deploying a wallet there.